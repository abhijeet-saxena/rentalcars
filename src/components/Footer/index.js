import React from 'react';
import { Box, Link } from '../../common/components';
import { FooterContainer, LinksContainer, ImagesContainer } from './styles';

import booking from '../../assets/images/booking.png';
import priceline from '../../assets/images/priceline.png';
import kayak from '../../assets/images/kayak.png';
import agoda from '../../assets/images/agoda.png';
import rentalcars from '../../assets/images/rentalcars.png';
import opentable from '../../assets/images/opentable.png';

const Footer = () => {
    return (
        <FooterContainer>
            <LinksContainer>
                <Link href="#" padding="0.5rem 0">
                    Privacy Notice
                </Link>
                <Link href="#" padding="0.5rem 0">
                    Cookies
                </Link>
                <Link href="#" padding="0.5rem 0">
                    Terms & Conditions
                </Link>
                <Link href="#" padding="0.5rem 0">
                    Help
                </Link>
                <Link href="#" padding="0.5rem 0">
                    Modern Slavery Statement
                </Link>
                <Link href="#" padding="0.5rem 0">
                    Supply Partner Enquiry and Marketplace
                </Link>
                <Link href="#" padding="0.5rem 0">
                    Affiliate Programme
                </Link>
                <Link href="#" padding="0.5rem 0">
                    Careers
                </Link>
            </LinksContainer>
            <Box width="80%" margin="1rem 0">
                Rentalcars.com is a trading name of Booking.com Transport
                Limited which is a limited company registered in England and
                Wales (Number: 05179829) whose registered address is at 100 New
                Bridge Street, London, EC4V 6JA. VAT number: GB 855349007.
                <br />
                <br />
                Rentalcars.com is part of Booking Holdings Inc., the world
                leader in online travel &amp; related services.
                <br />
                <br />
                Booking.com Transport Limited (FCA Firm Reference No 777648) is
                an Appointed Representative of Cover Genius Limited, which is
                authorised and registered by the Financial Conduct Authority,
                Firm Reference No 750711. Traveljigsaw Insurance Limited is
                incorporated in Malta and is authorised by the Malta Financial
                Services Authority to carry on the business of insurance in
                terms of the Insurance Business Act 1998. Company Registration
                Number: C82173 Registered Office, Development House, St Anne
                Street, Floriana, FRN 9010, Malta.
                <br />
                <br />
                <ImagesContainer>
                    <img src={booking} alt="booking" width="91px"></img>
                    <img src={priceline} alt="priceline" width="91px"></img>
                    <img src={kayak} alt="kayak" width="79px"></img>
                    <img src={agoda} alt="agoda" width="70px"></img>
                    <img src={rentalcars} alt="rentalcars" width="109px"></img>
                    <img src={opentable} alt="opentable" width="95px"></img>
                </ImagesContainer>
                <br />
                Copyright Â© 2020 Booking.com Transport Limited trading as
                Rentalcars.com. All rights reserved
            </Box>
        </FooterContainer>
    );
};

export default Footer;
