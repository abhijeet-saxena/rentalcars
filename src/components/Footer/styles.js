import { Flex, colors } from '../../common/components';
import styled from 'styled-components';

export const FooterContainer = styled(Flex)`
    flex-direction: column;
    background: ${colors.backgroundGray};
    text-align: center;
    align-items: center;
    font-size: 14px;
    padding-bottom: 2rem;
`;

export const LinksContainer = styled(Flex)`
    flex-direction: column;
    align-items: center;
    justify-content: center;
    background: ${colors.primaryBlue};
    padding: 1rem;
    > a {
        margin: 0.5rem;
    }
    @media (min-width: 768px) {
        flex-direction: row;
    }
`;

export const ImagesContainer = styled(Flex)`
    justify-content: space-between;
    width: 70vw;
    margin: 0 auto;
    @media (max-width: 768px) {
        display: none;
    }
`;
