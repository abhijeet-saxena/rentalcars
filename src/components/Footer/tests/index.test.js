import React from 'react';
import { render, cleanup } from '@testing-library/react';
import Footer from '../index';

describe('<Footer />', () => {
    afterAll(cleanup);
    it('Rendered Snapshot Matches', () => {
        const footer = render(<Footer />);
        expect(footer).toMatchSnapshot();
    });

    test('Site Links Are Displayed', () => {
        const { getByTestId } = render(<Footer />);
        expect(getByTestId('site-links')).toBeInTheDocument();

        expect(getByTestId('site-links').querySelector('a')).toHaveAttribute(
            'href',
            'https://www.rentalcars.com/PrivacyPolicy.do',
        );
    });

    test('Copyright text is Displayed', () => {
        const { getByTestId } = render(<Footer />);

        expect(
            getByTestId('copyright').querySelector('section'),
        ).toHaveTextContent('Rentalcars.com is a trading name of Booking.com');
    });

    test('Footer image is Displayed', () => {
        const { getByTestId } = render(<Footer />);

        expect(getByTestId('copyright').querySelector('img')).toHaveAttribute(
            'src',
            'https://r-cf.bstatic.com/static/img/tfl/group_logos/logo_booking/27c8d1832de6a3123b6ee45b59ae2f81b0d9d0d0.png',
        );
    });
});
