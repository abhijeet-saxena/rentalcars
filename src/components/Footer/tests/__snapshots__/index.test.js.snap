// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Footer /> Rendered Snapshot Matches 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <footer
        class="sc-gKAaRy dQESZe"
      >
        <ul
          class="sc-iCoGMd joATZ"
          data-testid="site-links"
        >
          <li>
            <a
              class="sc-dlnjwi jCSEIT"
              color="#FFF"
              href="https://www.rentalcars.com/PrivacyPolicy.do"
            >
              Privacy Notice
            </a>
          </li>
          <li>
            <a
              class="sc-dlnjwi jCSEIT"
              color="#FFF"
              href="https://www.rentalcars.com/TermsAndConditions.do"
            >
              Terms & Conditions
            </a>
          </li>
          <li>
            <a
              class="sc-dlnjwi jCSEIT"
              color="#FFF"
              href="https://www.rentalcars.com/Help.do"
            >
              Help
            </a>
          </li>
          <li>
            <a
              class="sc-dlnjwi jCSEIT"
              color="#FFF"
              href="https://cdn2.rcstatic.com/documentation/pdf/SignedModernSlaveryStatement.pdf"
            >
              Modern Slavery Statement
            </a>
          </li>
          <li>
            <a
              class="sc-dlnjwi jCSEIT"
              color="#FFF"
              href="https://secure.rentalcars.com/marketplace/en"
            >
              Supply Partner Enquiry and Marketplace
            </a>
          </li>
          <li>
            <a
              class="sc-dlnjwi jCSEIT"
              color="#FFF"
              href="https://www.rentalcarsconnect.com/"
            >
              Affiliate Programme
            </a>
          </li>
          <li>
            <a
              class="sc-dlnjwi jCSEIT"
              color="#FFF"
              href="https://careers.booking.com/"
            >
              Careers
            </a>
          </li>
           
        </ul>
        <div
          class="sc-bdnxRM sc-fujyAs hoHvTM hiegXk"
          data-testid="copyright"
        >
          <section>
            Rentalcars.com is a trading name of Booking.com Transport Limited which is a limited company registered in England and Wales (Number: 05179829) whose registered address is at 100 New Bridge Street, London, EC4V 6JA. VAT number: GB 855349007.
          </section>
          <section>
            Rentalcars.com is part of Booking Holdings Inc., the world leader in online travel & related services.
          </section>
          <section>
            Booking.com Transport Limited (FCA Firm Reference No 777648) is an Appointed Representative of Cover Genius Limited, which is authorised and registered by the Financial Conduct Authority, Firm Reference No 750711. Traveljigsaw Insurance Limited is incorporated in Malta and is authorised by the Malta Financial Services Authority to carry on the business of insurance in terms of the Insurance Business Act 1998. Company Registration Number: C82173 Registered Office, Development House, St Anne Street, Floriana, FRN 9010, Malta.
          </section>
          <div
            class="sc-bdnxRM sc-pNWdM hoHvTM iUCHmb"
          >
            <img
              alt=""
              height="26px"
              src="https://r-cf.bstatic.com/static/img/tfl/group_logos/logo_booking/27c8d1832de6a3123b6ee45b59ae2f81b0d9d0d0.png"
              width="91px"
            />
            <img
              alt=""
              height="26px"
              src="https://q-cf.bstatic.com/static/img/tfl/group_logos/logo_priceline/f80e129541f2a952d470df2447373390f3dd4e44.png"
              width="91px"
            />
            <img
              alt=""
              height="26px"
              src="https://r-cf.bstatic.com/static/img/tfl/group_logos/logo_kayak/83ef7122074473a6566094e957ff834badb58ce6.png"
              width="79px"
            />
            <img
              alt=""
              height="26px"
              src="https://q-cf.bstatic.com/static/img/tfl/group_logos/logo_agoda/1c9191b6a3651bf030e41e99a153b64f449845ed.png"
              width="70px"
            />
            <img
              alt=""
              height="26px"
              src="https://r-cf.bstatic.com/static/img/tfl/group_logos/logo_rentalcars/6bc5ec89d870111592a378bbe7a2086f0b01abc4.png"
              width="109px"
            />
            <img
              alt=""
              height="26px"
              src="https://r-cf.bstatic.com/static/img/tfl/group_logos/logo_opentable/a4b50503eda6c15773d6e61c238230eb42fb050d.png"
              width="95px"
            />
          </div>
          <section>
            Copyright © 2020 Booking.com Transport Limited trading as Rentalcars.com. All rights reserved
          </section>
        </div>
      </footer>
    </div>
  </body>,
  "container": <div>
    <footer
      class="sc-gKAaRy dQESZe"
    >
      <ul
        class="sc-iCoGMd joATZ"
        data-testid="site-links"
      >
        <li>
          <a
            class="sc-dlnjwi jCSEIT"
            color="#FFF"
            href="https://www.rentalcars.com/PrivacyPolicy.do"
          >
            Privacy Notice
          </a>
        </li>
        <li>
          <a
            class="sc-dlnjwi jCSEIT"
            color="#FFF"
            href="https://www.rentalcars.com/TermsAndConditions.do"
          >
            Terms & Conditions
          </a>
        </li>
        <li>
          <a
            class="sc-dlnjwi jCSEIT"
            color="#FFF"
            href="https://www.rentalcars.com/Help.do"
          >
            Help
          </a>
        </li>
        <li>
          <a
            class="sc-dlnjwi jCSEIT"
            color="#FFF"
            href="https://cdn2.rcstatic.com/documentation/pdf/SignedModernSlaveryStatement.pdf"
          >
            Modern Slavery Statement
          </a>
        </li>
        <li>
          <a
            class="sc-dlnjwi jCSEIT"
            color="#FFF"
            href="https://secure.rentalcars.com/marketplace/en"
          >
            Supply Partner Enquiry and Marketplace
          </a>
        </li>
        <li>
          <a
            class="sc-dlnjwi jCSEIT"
            color="#FFF"
            href="https://www.rentalcarsconnect.com/"
          >
            Affiliate Programme
          </a>
        </li>
        <li>
          <a
            class="sc-dlnjwi jCSEIT"
            color="#FFF"
            href="https://careers.booking.com/"
          >
            Careers
          </a>
        </li>
         
      </ul>
      <div
        class="sc-bdnxRM sc-fujyAs hoHvTM hiegXk"
        data-testid="copyright"
      >
        <section>
          Rentalcars.com is a trading name of Booking.com Transport Limited which is a limited company registered in England and Wales (Number: 05179829) whose registered address is at 100 New Bridge Street, London, EC4V 6JA. VAT number: GB 855349007.
        </section>
        <section>
          Rentalcars.com is part of Booking Holdings Inc., the world leader in online travel & related services.
        </section>
        <section>
          Booking.com Transport Limited (FCA Firm Reference No 777648) is an Appointed Representative of Cover Genius Limited, which is authorised and registered by the Financial Conduct Authority, Firm Reference No 750711. Traveljigsaw Insurance Limited is incorporated in Malta and is authorised by the Malta Financial Services Authority to carry on the business of insurance in terms of the Insurance Business Act 1998. Company Registration Number: C82173 Registered Office, Development House, St Anne Street, Floriana, FRN 9010, Malta.
        </section>
        <div
          class="sc-bdnxRM sc-pNWdM hoHvTM iUCHmb"
        >
          <img
            alt=""
            height="26px"
            src="https://r-cf.bstatic.com/static/img/tfl/group_logos/logo_booking/27c8d1832de6a3123b6ee45b59ae2f81b0d9d0d0.png"
            width="91px"
          />
          <img
            alt=""
            height="26px"
            src="https://q-cf.bstatic.com/static/img/tfl/group_logos/logo_priceline/f80e129541f2a952d470df2447373390f3dd4e44.png"
            width="91px"
          />
          <img
            alt=""
            height="26px"
            src="https://r-cf.bstatic.com/static/img/tfl/group_logos/logo_kayak/83ef7122074473a6566094e957ff834badb58ce6.png"
            width="79px"
          />
          <img
            alt=""
            height="26px"
            src="https://q-cf.bstatic.com/static/img/tfl/group_logos/logo_agoda/1c9191b6a3651bf030e41e99a153b64f449845ed.png"
            width="70px"
          />
          <img
            alt=""
            height="26px"
            src="https://r-cf.bstatic.com/static/img/tfl/group_logos/logo_rentalcars/6bc5ec89d870111592a378bbe7a2086f0b01abc4.png"
            width="109px"
          />
          <img
            alt=""
            height="26px"
            src="https://r-cf.bstatic.com/static/img/tfl/group_logos/logo_opentable/a4b50503eda6c15773d6e61c238230eb42fb050d.png"
            width="95px"
          />
        </div>
        <section>
          Copyright © 2020 Booking.com Transport Limited trading as Rentalcars.com. All rights reserved
        </section>
      </div>
    </footer>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
